homeassistant/components/switch/unifi.py

```diff
@@ -0,0 +1,221 @@
+                new_switches.append(switches[client_id])
+                LOGGER.debug("New UniFi switch %s (%s)",
+                             client.hostname, client.mac)
+
+        elif client_id not in progress_waiting:
+            LOGGER.debug("Updating UniFi switch %s (%s)",
+                         switches[client_id].entity_id,
+                         switches[client_id].client.mac)
+            switches[client_id].async_schedule_update_ha_state()
+
+    update_storage = False
+    for device in controller.api.devices.values():
+        for port in device.ports.values():
+
+            if port.port_poe and port.poe_mode == 'off':
+                for client in controller.api.clients.values():
+                    if not client.mac in switches_off and \
+                       device.mac == client.sw_mac and \
+                       port.port_idx == client.sw_port:
+                        switches_off[client.mac] = client.raw
+                        update_storage = True
+
+            elif port.port_poe and port.poe_mode is not None:
+                for client in controller.api.clients.values():
+                    if client.mac in switches_off and \
+                       device.mac == client.sw_mac and \
+                       port.port_idx == client.sw_port:
+                        del switches_off[client.mac]
+                        update_storage = True
+    @callback
+    def _data_to_save():
+        """Update storage with devices that has got its' POE turned off."""
+        return switches_off
+    if update_storage:
+        store.async_delay_save(_data_to_save, SAVE_DELAY)
+
+    if new_switches:
+        async_add_entities(new_switches)
+
+
+class UniFiSwitch(SwitchDevice):
+    """Representation of a client that uses POE."""
+
+    def __init__(self, client, controller, request_controller_update):
+        """Set up switch."""
+        self.client = client
+        self.controller = controller
+        self.async_request_controller_update = request_controller_update
+
+    async def async_update(self):
+        """Synchronize state with controller."""
+        await self.async_request_controller_update(self.client.mac)
+
+    @property
+    def name(self):
+        """Return the name of the switch."""
+        return self.client.hostname
+
+    @property
+    def unique_id(self):
+        """Return a unique identifier for this switch."""
+        return 'poe-{}'.format(self.client.mac)
+
+    @property
+    def is_on(self):
+        """Return true if POE is active."""
+        return self.port.poe_mode != 'off'
+
+    @property
+    def available(self):
+        """Return if switch is available."""
+        return self.controller.available or \
+            self.client.sw_mac in self.controller.api.devices
+
+    async def async_turn_on(self, **kwargs):
+        """Enable POE for client."""
+        device_id = self.device.id
+        url = 's/{site}/rest/device/' + device_id
```

Comment:
This looks like it should be part of the aiounifi library.

Reply:
Indeed! Thanks!